index.bs

```diff
@@ -438,32 +431,119 @@ partial interface NeuralNetworkContext {
+    value of *layout*.
+
+    <div class="note">
+    A "global" pooling operation such as the *globalMaxPool2d* operation is a variant of the pooling operation where the window dimensions is the dimensions of the input as follow.
+    <pre highlight="js">
+    Operand globalMaxPool2d(input) {
+      return maxPool2d(input, input.shape());
+    }
+    </pre>
+    </div>
+</div>
+
+### reduce ### {#api-neuralnetworkcontext-reduce}
+Reduce the input along the dimensions given in *axes*.
+<script type=idl>
+partial interface NeuralNetworkContext {
+  Operand reduceL1(Operand input, optional sequence<long> axes = null, optional boolean keepDimensions = false);
+  Operand reduceL2(Operand input, optional sequence<long> axes = null, optional boolean keepDimensions = false);
+  Operand reduceLogSum(Operand input, optional sequence<long> axes = null, optional boolean keepDimensions = false);
+  Operand reduceLogSumExp(Operand input, optional sequence<long> axes = null, optional boolean keepDimensions = false);
+  Operand reduceMax(Operand input, optional sequence<long> axes = null, optional boolean keepDimensions = false);
+  Operand reduceMean(Operand input, optional sequence<long> axes = null, optional boolean keepDimensions = false);
+  Operand reduceMin(Operand input, optional sequence<long> axe = null, optional boolean keepDimensions = false);
+  Operand reduceProduct(Operand input, optional sequence<long> axes = null, optional boolean keepDimensions = false);
+  Operand reduceSum(Operand input, optional sequence<long> axes = null, optional boolean keepDimensions = false);
+  Operand reduceSumSquare(Operand input, optional sequence<long> axes = null, optional boolean keepDimensions = false);
+};
+</script>
+<div algorithm=reduce>
+    **Arguments:**
+        - *input*: an {{Operand}}. The input tensor.
+        - *axes*: an optional sequence of {{long}}. The dimensions to reduce.
+            If not present, all dimensions are reduced.
+        - *keepDimensions*: an optional {{boolean}}. If true, retains reduced dimensions with size of 1.
+            The default value is false.
+
+    **Returns:** an {{Operand}}. The reduced output tensor.
+
+    **Reduction types:**
+        - *L1*: Compute the <a href="https://mathworld.wolfram.com/L1-Norm.html">L1 norm</a> of the input elements along the axes.
+        - *L2*: Compute the <a href="https://mathworld.wolfram.com/L2-Norm.html">L2 norm</a> of the input elements along the axes.
+        - *LogSum*: Compute the `log(sum(input values))` of the input values along the axes.
+        - *Max*: Compute the `max(input values)` of the input values along the axes.
+        - *Mean*: Compute the `mean(input values)` of the input values along the axes.
+        - *Min*: Compute the `min(input values)` of the input values along the axes.
+        - *Product*: Compute the product of the input values along the axes.
+        - *Sum*: Compute the `sum(input values)` of the input values along the axes.
```

Comment:
Should we define the function `sum(x)`? Or as Product, say "Compute the sum of ...". Same for min, max, mean and log.

Reply:
My personal preference is to use the style used for "Product" for everything.

Reply:
Fixed. Please take a look at my new commit.